code/game/objects/mail.dm

```diff
@@ -0,0 +1,233 @@
+		return
+	if(do_after(user, 1 SECONDS, target = user) && !QDELETED(src))
+		to_chat(user, "<span class='notice'>You begin to open the envelope.</span>")
+		playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+		user.unEquip(src)
+		for(var/obj/item/I in contents)
+			user.put_in_hands(I)
+		qdel(src)
+
+/obj/item/envelope/Initialize(mapload)
+	. = ..()
+	var/item = pick(possible_contents)
+	new item(src)
+	new /obj/item/stack/spacecash(src, rand(1, 50) * 5)
+	var/list/mind_copy = SSticker.minds.Copy()
+	shuffle(mind_copy)
+	for(var/datum/mind/mail_attracted_people in mind_copy)
+		var/turf/T = get_turf(mail_attracted_people)
+		if(mail_attracted_people.offstation_role || !ishuman(mail_attracted_people) || T.z == 1)
+			continue
+		if(mail_attracted_people.assigned_role in job_list)
+			recipient = mail_attracted_people.current
+			name = "letter to [recipient.real_name]"
+			return
+	if(!admin_spawned)
+		log_debug("Error: failed to find a new name to assign to [src]!")
+		qdel(src)
+
+/obj/item/envelope/security
+	icon_state = "mail_sec"
+	possible_contents = list(/obj/item/reagent_containers/food/snacks/donut/sprinkles,
+							/obj/item/megaphone,
+							/obj/item/poster/random_official,
+							/obj/item/restraints/handcuffs/pinkcuffs,
+							/obj/item/restraints/legcuffs/bola/energy,
+							/obj/item/reagent_containers/food/drinks/coffee,
+							/obj/item/stock_parts/cell/super,
+							/obj/item/grenade/barrier/dropwall,
+							/obj/item/toy/figure/crew/detective,
+							/obj/item/toy/figure/crew/hos,
+							/obj/item/toy/figure/crew/secofficer)
+	job_list = list("Head of Security", "Security Officer", "Detective", "Warden")
+
+/obj/item/envelope/science
+	icon_state = "mail_sci"
+	possible_contents = list(/obj/item/analyzer,
+	/obj/item/assembly/signaler,
+	/obj/item/slime_extract/grey,
+	/obj/item/clothing/mask/gas,
+	/obj/item/reagent_containers/spray/cleaner,
+	/obj/item/clothing/glasses/regular,
+	/obj/item/taperecorder,
+	/obj/item/paicard,
+	/obj/item/toy/figure/crew/borg,
+	/obj/item/toy/figure/crew/geneticist,
+	/obj/item/toy/figure/crew/rd,
+	/obj/item/toy/figure/crew/roboticist,
+	/obj/item/toy/figure/crew/scientist)
+	job_list = list("Research Director", "Roboticist", "Geneticist",  "Scientist")
+
+/obj/item/envelope/supply
+	icon_state = "mail_sup"
+	possible_contents = list(/obj/item/reagent_containers/hypospray/autoinjector/survival,
+	/obj/item/reagent_containers/food/drinks/bottle/absinthe/premium,
+	/obj/item/clothing/glasses/meson/gar,
+	/obj/item/stack/marker_beacon/ten,
+	/obj/item/stack/medical/splint,
+	/obj/item/pen/multi/fountain,
+	/obj/item/clothing/mask/cigarette/cigar,
+	/obj/item/stack/wrapping_paper,
+	/obj/item/toy/figure/crew/cargotech,
+	/obj/item/toy/figure/crew/qm,
+	/obj/item/toy/figure/crew/miner)
+	job_list = list("Quartermaster", "Cargo Technician", "Shaft Miner")
```

Comment:
Couldnt find a way to check for the jobs without manually adding it here?

Reply:
See my below comment.

Reply:
What the comment below has to do with it?